File: ruby_src/lib/limelight/controllers/textbox.rb
module Limelight
  module Controllers
    
    module Textbox
      class << self
        def extended(block)
          block.panel.painters.clear
          block.panel.painters << Java.limelight.TextboxPainter.new(block.panel)
          block.panel.clear_event_listeners
          set_default_styles(block)
        end
        
        def set_default_styles(block)
          block.style.width = "120"
          block.style.height = "22"
        end
      end
  
    end
  
  end
end

File: src/limelight/ui/model/inputs/TextBox.java
package limelight.ui.model.inputs;

import limelight.util.Box;
import limelight.ui.model.updates.Updates;
import limelight.ui.model.updates.BoundedPaintUpdate;
import limelight.ui.model.updates.BoundedShallowPaintUpdate;

import javax.swing.*;
import java.awt.*;

public class TextBox extends JTextField
{
  private TextBoxPanel panel;

  public TextBox(TextBoxPanel panel)
  {
    this.panel = panel;
  }

  public void repaint()
  {
    if(panel != null)
      panel.setNeededUpdate(Updates.paintUpdate);
  }

  public void repaint(long tm, int x, int y, int width, int height)
  {
    if(panel != null)
      panel.setNeededUpdate(new BoundedPaintUpdate(x, y, width, height));
  }

  public void repaint(Rectangle r)
  {
    if(panel != null)
      panel.setNeededUpdate(new BoundedPaintUpdate(r));
  }

  public boolean isShowing()
  {
    return true;
  }
}
